src/main/java/com/redhat/cloud/notifications/NotificationsGwApp.java

```diff
@@ -67,10 +67,13 @@ private String readGitProperties() {
         }
     }
 
-    private String readFromInputStream(InputStream inputStream)
-            throws IOException {
+    private String readFromInputStream(InputStream inputStream) throws IOException {
+        if(inputStream == null) {
+            return "git.properties file not available";
+        }
         StringBuilder resultStringBuilder = new StringBuilder();
-        try (BufferedReader br = new BufferedReader(new InputStreamReader(inputStream))) {
+        final InputStreamReader inputStreamReader = new InputStreamReader(inputStream);
+        try (BufferedReader br = new BufferedReader(inputStreamReader)) {
```

Comment:
I don't know if `InputStreamReader` can throw an exception (other than NPE) when its constructor is called. But if it does, this would ensure the underlying `InputStream` is correctly closed:

```suggestion
        try (InputStreamReader inputStreamReader = new InputStreamReader(inputStream); BufferedReader br = new BufferedReader(inputStreamReader)) {
```